examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/OverviewFragment.kt

```diff
@@ -103,6 +116,52 @@ class OverviewFragment : Fragment(), OfferingCardAdapter.OfferingCardAdapterList
         findNavController().navigate(directions, extras)
     }
 
+    private fun showPurchaseProductIdDialog() {
+        val builder = MaterialAlertDialogBuilder(requireContext())
+        builder.setTitle("Enter the Product ID you want to purchase:")
+        val input = EditText(context)
+        input.inputType = InputType.TYPE_CLASS_TEXT
+        builder.setView(input)
+        builder.setPositiveButton("Purchase") { dialog, which ->
+            val productId = input.text.toString()
+            Purchases.sharedInstance.getProducts(
+                listOf(productId),
+                object : GetStoreProductsCallback {
+                    override fun onReceived(storeProducts: List<StoreProduct>) {
+                        if (storeProducts.isEmpty()) {
+                            showError("A product with ID $productId does not exist")
+                            return
+                        }
+                        Purchases.sharedInstance.purchase(
+                            PurchaseParams.Builder(requireActivity(), storeProducts.first()).build(),
```

Comment:
Instead of `storeProducts.first()`, can this prompt try to split by `:` to also look for a matching base plan?